packages/bitcore-wallet-service/src/lib/chain/xrp/index.ts

```diff
@@ -133,24 +133,30 @@ export class XrpChain implements IChain {
   }
 
   getBitcoreTx(txp, opts = { signed: true }) {
-    const { destinationTag, outputs } = txp;
+    const { destinationTag, outputs, outputOrder, multiTx } = txp;
     const chain = 'XRP';
-    const recipients = outputs.map(output => {
-      return {
-        amount: output.amount,
-        address: output.toAddress,
-        tag: output.tag
-      };
-    });
     const unsignedTxs = [];
-    for (let index = 0; index < recipients.length; index++) {
-      const _tag = recipients[0]?.tag || destinationTag;
+    const length = multiTx ? outputOrder.length : outputs.length;
```

Comment:
Why is there a difference between outputs.length and outputOrder.length?

For XRP, can't we infer that `multiTx` is true if output[s/Order].length > 1 instead of explicitly passing in `multiTx`? Is there ever a case where an XRP txp has outpus.length > 1 and we _wouldn't_ send out multiple txs?

Reply:
outputOrder previously was used strictly for utxo currencies, for a case that is not clear to me. it did some increments during txp creation that made  outputs.length != outputOrder.length. I used the same variable because the name makes sense. from original txproposal.ts:

```
let numOutputs = x.outputs.length + 1;
if (x.instantAcceptanceEscrow) {
  numOutputs = numOutputs + 1;
}
x.outputOrder = _.range(numOutputs);
```

You are correct in stating XRP txp outpus.length > 1  would always be a multiTx. to me it was cleaner to set and check that variable  instead of checking txp.chain == 'xrp' && txp.outpus.length > 1 in multiple places (see last xrp payout bitcore pr). also would port well if we support any other chain that needs this functionality (god forbid)

Reply:
Ah! So, I can give some background on outputOrder. It's meant to randomize the outputs so the change address is not in a predictable position, thus enhancing privacy. And speaking of the change address, outputs.length != outputOrder.length for UTXO txps because the change output is reflected as the changeAddress on the txp instead of being in the outputs array. Since there's no concept as change in XRP land, outputs.length should always equal outputOrder.length.

I think keeping a multiTx flag on the txp is fine... :thinking:  The only thing to consider may be from the perspective of other currencies. A) it might be tempting for someone to set that as true for a UTXO or EVM txp, for example, when what they should really be doing is creating multiple txps. B) it might be confusing to someone else who might wonder "why would I ever want that? Is this a use case I need to consider for X currency?". In those cases, keeping multiTx implicit on the schema and having a helper fn that explains when a txp should be multiTx would be better.